File: MyWebServices/Core/DataAccess/Entities/CustomUserElement.cs
namespace MyWebServices.Core.DataAccess.Entities
{
    public class CustomUserElement
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Value { get; set; }
        public string? TemplateValue { get; set; }
        public SortingOrder ElementSortingOrder { get; set; }
        public int? UserPatternId { get; set; }
        public int? UserSettingsEntityId { get; set; }

        public enum SortingOrder
        {
            BeforeText,
            AfterText,
            Template
        }
    }
}

File: MyWebServices/ClientApp/src/app/home/CustomUserElement.ts
import {SortingOrder} from "./SortingOrder";

export class CustomUserElement {
  public id: number = 0;
  public name: string = '';
  public value: string = '';
  public templateValue: string | null = null;
  public elementSortingOrder: number = 0;
  public userPatternId: number | null = null;

  public withoutTemplate: boolean = (this.templateValue == null || this.templateValue?.length === 0 );
}

